src/flib/pionfput_mod.F90.in

```diff
@@ -111,22 +111,48 @@ contains
     character, allocatable :: cval(:)
     integer :: i
     integer, allocatable :: count(:)
-    integer :: ndims
+    integer :: ndims, clen
+
+    if(len(ival) == 0) then
+      print *, "PIO: WARNING: Empty string passed to PIO_put_var,",&
+                " function put_var1_text, a collective C API call will",&
+                " be skipped on this process. ", __PIO_FILE__, __LINE__
+    end if
+
+    ierr = get_text_var_sz(File, varid, clen)
+    if(ierr /= PIO_NOERR) then
+      call piodie(__PIO_FILE__, __LINE__, "Getting text variable size failed")
+    end if
+
+    if(clen < len_trim(ival)) then
+      print *, "PIO: WARNING: The length of the string in the user buffer,",&
+                len_trim(ival),&
+                " chars, is greater than the size of the variable",&
+                " being written to, ", clen, " chars. The string",&
+                " may be truncated"
```

Comment:
We can add file and line info for this warning:
`" may be truncated" ==> " may be truncated. ", __PIO_FILE__, __LINE__`